lib/modules/datasource/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  type AddonInfo,
+  type Compatibility,
+  DescribeAddonVersionsCommand,
+  EKSClient,
+} from '@aws-sdk/client-eks';
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import * as awsEksAddonVersioning from '../../versioning/aws-eks-addon';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { EksAddonsFilter } from './schema';
+
+export class AwsEKSAddonDataSource extends Datasource {
+  static readonly id = 'aws-eks-addon';
+
+  override readonly defaultVersioning = awsEksAddonVersioning.id;
+  override readonly caching = true;
+  private readonly clients: Record<string, EKSClient> = {};
+
+  override readonly defaultConfig: Record<string, unknown> | undefined = {
+    commitMessageTopic: '{{datasource}}',
+    commitMessageExtra: '{{currentVersion}} to {{{newVersion}}}',
+    additionalBranchPrefix: '{{datasource}}-{{{newVersion}}}-',
```

Comment:
sorry, but I forgot why this is necessary 🤔

Reply:
I could remove them, but just to make it a bit nicer for branches and messages.

Comment https://github.com/renovatebot/renovate/pull/33272#issuecomment-2628560903

<img width="1325" alt="Screenshot 2025-02-24 at 17 12 13" src="https://github.com/user-attachments/assets/9fe5a206-be87-4f8e-930d-c60e8f433a53" />

For cases when 

```
eksAddonsFilter={"addonName":"kube-proxy", "default":true, "kubernetesVersion":"1.30"}
```

Renovate could not proper handle titles, branches. Would you recommend to remove this completely?

Reply:
how is it handled in other AWS datasource? I assume they've a pretty depType?

Reply:
Side note, `aws-rds` produces incomprehencable branch name and commit message. Users are effectively forced to override them to something sensible, see https://github.com/renovatebot/renovate/discussions/32560

Reply:
Using additionalBranchPrefix here is highly undesirable.

It would be better overall if "depName" could be set to something human readable

Reply:
I agree. How could I do that? Feels like some shared method is needed or there is already a way?

I'm thinking to remove that `defaultConfig` entirely and fine if renovate could resolve it for us. 

Please share desired path

Reply:
`depName` is set by managers during the extract phase. If this datasource is only used with custom datasources, then it could/should be set there and isn't necessary for the datasource to massage it. The `packageName` is what's used for the actual lookup.

Reply:
Just FIY, the package name in such case is the object

```
eksAddonsFilter={"kubernetesVersion":"1.30","addonName":"vpc-cni"}
```

So this creates PRs with branches, titles and etc that do contain special characters. I have no opinion on that.

Reply:
Any massaging or normalization of packageName could/would break things. If you mean normalizing packageName into a prettier depName, that's the job of the manager extract.

Reply:
This is a custom regex datasource. Implementing manager extract is another journey from a first look. If possible I would like to keep it simple. I'm not familiar with code in `modules/manager/...`.

Reply:
Whoever writes the custom manager definition can also define depName at the same time. No renovate code changes required for that